go/cmd/vtctldclient/command/keyspace_routing_rules.go

```diff
@@ -105,11 +105,16 @@ func commandApplyKeyspaceRoutingRules(cmd *cobra.Command, args []string) error {
 				fmt.Printf(" in the following cells: %s.\n", strings.Join(applyKeyspaceRoutingRulesOptions.Cells, ", "))
 			}
 		}
-
 		return nil
 	}
 
-	_, err = client.ApplyKeyspaceRoutingRules(commandCtx, &vtctldatapb.ApplyKeyspaceRoutingRulesRequest{
+	// Get current keyspace routing rules to include in the response.
+	currentRules, err := client.GetKeyspaceRoutingRules(commandCtx, &vtctldatapb.GetKeyspaceRoutingRulesRequest{})
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.ApplyKeyspaceRoutingRules(commandCtx, &vtctldatapb.ApplyKeyspaceRoutingRulesRequest{
```

Comment:
We currently have an inconsistency in how we treat `ApplyKeyspaceRoutingRulesResponse`. This RPC always returns an empty response. It feels weird to add fields to the RPC response, which the RPC doesn't actually return, but which we stuff in for the purpose of JSON-inification before printing it out.
I understand that the reason we don't want to include these in the RPC response itself is because it can potentially be huge. Given that though, we should remove the newly added fields from the proto, and add a bit more code to this function to print the old and new rules separately. Putting them into the response seems to be a convenience.

Reply:
> We currently have an inconsistency in how we treat `ApplyKeyspaceRoutingRulesResponse`. This RPC always returns an empty response. It feels weird to add fields to the RPC response, which the RPC doesn't actually return, but which we stuff in for the purpose of JSON-inification before printing it out. I understand that the reason we don't want to include these in the RPC response itself is because it can potentially be huge. Given that though, we should remove the newly added fields from the proto, and add a bit more code to this function to print the old and new rules separately. Putting them into the response seems to be a convenience.

Ah, right. I should have sent it in the RPC itself. Is it fine to move that to the RPC? That way we have the functionality of old/new rules also if a client makes an RPC call rather than cli.

Reply:
Yes sure, that will be fine.

Reply:
Done: https://github.com/vitessio/vitess/pull/15807/commits/f483725cba307b999311a6dbe4f97d6470dfbef8